src/Bicep.Core/Syntax/TextNode.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax
+{
+    public class TextNode
```

Comment:
I think it would be worth using the `Token` class directly in `DisableNextLineDiagnosticsSyntaxTrivia`, rather than creating a new token-like thing.

I did some digging through the Roslyn code, and it looks like that's the path they take:

* SyntaxToken (extends CsharpSyntaxNode, contains leading & trailing trivia, similar to our setup): https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxToken.cs#L16
* StructuredTriviaSyntax extends CsharpSyntaxNode: https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Syntax/StructuredTriviaSyntax.cs#L12
* DirectiveTriviaSyntax extends StructuredTriviaSyntax: https://github.com/dotnet/roslyn/blob/96d6017c378ab311544c5521a3e5c6616fea48a5/src/Compilers/CSharp/Portable/Generated/CSharpSyntaxGenerator/CSharpSyntaxGenerator.SourceGenerator/Syntax.xml.Syntax.Generated.cs#L14685

Reply:
Thanks Anthony. I updated to use Token instead of TextNode and it works well.
https://github.com/Azure/bicep/commit/70a1943c8a863af2c071d6f439f3da61bfa11844

I am not creating leading and trailing trivia for the Token. Let me know if that's fine.